cluster-install/install.sh

```diff
@@ -214,12 +229,33 @@ load_config_from_file_if_given() {
         add_problem "CONFIG_FILE '$config_file' does not exist" "Given CONFIG_FILE '$config_file' does not exist"
         return 1
     fi
-
-    # shellcheck disable=SC1090
-    source "$config_file"
+    # Store the current env so it can be re-applied after sourcing the config file
+    # Note: we do some ugly looking bash magic here, because we want to double quote each value
+    # so they are sourced properly, but without potentially affecting the contents. Particularly,
+    # values containing '=' are tricky since that's also the assignment operator.
+    local previous_env_double_quoted=""
+    while IFS= read -r line; do
+        # Remove the longest match (two '%') of '=*' (equal sign + wildcard) from the end of the line ('%' as opposed
+        # to '#'). This ensures we only keep the key, without potentially catching an '=' that's part of the
+        # variable's value (such as in b64 encoded values).
+        key="${line%%=*}"
+
+        # This does something similar, but this time removing the shortest match (one '#') of '*='
+        # (wildcard + equal sign) from the beginning ('#' as opposed to '%'), keeping only the value.
+        value="${line#*=}"
+
+        # Rebuild the env with the double quotes
+        previous_env_double_quoted+="$key=\"$value\"${__NEWLINE}"
+    done < <(env)
```

Comment:
Getting the user's env was pretty easy (just call `env`), but unfortunately, that trimmed out any surrounding quotes from the values, causing issues when we re-applied them, so we have to re-quote the values ourselves.